src/test/java/net/barrage/school/java/ecatalog/app/HibernateExampleTest.java

```diff
@@ -56,8 +75,10 @@ interface ProductRepository extends CrudRepository<Product, UUID> {
 
 @Accessors(chain = true)
 @Data
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
```

Comment:
Set as an interface won't work correctly if we do not define equals and hashCode - https://www.baeldung.com/java-equals-hashcode-contracts